include/configs.hpp

```diff
@@ -1,3 +1,6 @@
+#ifndef CONFIGS_HEADER
```

Comment:
I think configs.hpp should be configs.h to keep consistent naming conventions. Unless there is some other reasoning behind the different extension

Reply:
I am actually moving all to hpp, but I haven't yet modified the others to justify renaming them yet.